railties/lib/rails/generators/rails/app/templates/Dockerfile.tt

```diff
@@ -9,15 +9,22 @@ WORKDIR /rails
 
 # Set production environment
 ENV RAILS_ENV="production" \
+    BUNDLE_PATH="vendor/bundle" \
     BUNDLE_WITHOUT="development"
 
+# Update gems and preinstall the desired version of bundler
+ARG BUNDLER_VERSION=<%= Bundler::VERSION %>
```

Comment:
I don't think we should copy the bundler version in the dockerfile. It's updated regularly, so people would need to keep it in sync with the Gemfile.lock. It's much simpler to let `bundle install` read the required version in `Gemfile.lock` and re-exec.